apis/topic/v1alpha1/topic_types.go

```diff
@@ -27,8 +27,91 @@ import (
 
 // TopicParameters are the configurable fields of a Topic.
 type TopicParameters struct {
+	// +kubebuilder:validation:Minimum:=1
 	ReplicationFactor int `json:"replicationFactor"`
+	// +kubebuilder:validation:Minimum:=1
 	Partitions int `json:"partitions"`
+	// +optional
```

Comment:
AFAIU, there is a clear separation with the above two parameter and the configurations below on Kafka side. 
So, it might be better to have the ones below under `Config` ?

I mean; 

```go
// TopicParameters are the configurable fields of a Topic.
type TopicParameters struct {
	// +kubebuilder:validation:Minimum:=1
	ReplicationFactor int `json:"replicationFactor"`
	// +kubebuilder:validation:Minimum:=1
	Partitions int `json:"partitions"`
        // +optional
        Config *TopicConfig `json:"config"`
}
```

Reply:
This was the option we went with. :)